src/compute/src/memory_management/memory_manager.rs

```diff
@@ -109,12 +110,25 @@ impl GlobalMemoryManager {
         let jemalloc_epoch_mib = jemalloc_epoch::mib().unwrap();
         let jemalloc_allocated_mib = jemalloc_stats::allocated::mib().unwrap();
 
-        let mut tick_interval =
-            tokio::time::interval(Duration::from_millis(self.barrier_interval_ms as u64));
+        let mut barrier_interval_ms =
+            Self::get_eviction_check_interval(&self.system_param_manager.get_params());
+        let mut tick_interval = tokio::time::interval(Duration::from_millis(barrier_interval_ms));
+        let mut params_rx = self.system_param_manager.watch_params();
 
         loop {
             // Wait for a while to check if need eviction.
-            tick_interval.tick().await;
+            select! {
+                // Barrier interval has changed.
+                Ok(_) = params_rx.changed() => {
+                    let new_barrier_interval_ms = Self::get_eviction_check_interval(&params_rx.borrow());
+                    if new_barrier_interval_ms != barrier_interval_ms
+                     {
+                        tick_interval = tokio::time::interval(Duration::from_millis(new_barrier_interval_ms));
+                        barrier_interval_ms = new_barrier_interval_ms;
+                    }
+                }
+                _ = tick_interval.tick() => {},
+            }
```

Comment:
Please don't change this. Make barrier interval too large may lead to oom.

Reply:
Shall we have another config for this interval?

Reply:
The eviction of the LRU cache (i.e. streaming memory control) actually still depends on barriers, so simply having another config for this checking interval may not help. If we do want to make barrier interval configurable, we should make sure our users know what they are doing (including the side effects). cc. @fuyufjh